iceoryx_examples/icediscovery_in_c/README.md

```diff
@@ -0,0 +1,134 @@
+# icediscovery in C
+
+## Introduction
+
+This example demonstrates how to search for specific services using iceoryx's
+service discovery. It provides two applications - one offering different
+services and one searching for those with different search queries. The
+behavior and structure is quite similar to the [icediscovery C++ example](https://github.com/eclipse-iceoryx/iceoryx/tree/v2.0.0/iceoryx_examples/icedelivery).
+
+<!--## Expected Output-->
+<!-- @todo Add expected output with asciinema recording before v2.0-->
+
+## Code walkthrough
+
+### Offer services
+
+We create several publisher ports which offer their services on construction by
+default. For more dynamism the `cameraPublishers` offer/stop their services
+periodically. If you want more information on how to create publisher ports,
+have a look at the [icedelivery C example](https://github.com/eclipse-iceoryx/iceoryx/tree/v2.0.0/iceoryx_examples/icedelivery_in_c).
+
+### Find services
+
+To be able to search for services, we need to include:
+
+<!--[geoffrey][iceoryx_examples/icediscovery_in_c/iox_c_find_service.c][include service discovery]-->
+```c
+#include "iceoryx_binding_c/service_discovery.h"
+```
+
+To get a handle to the service discovery we call `iox_service_discovery_init`.
+The service discovery requires memory of the amount of
+`sizeof(iox_service_discovery_storage_t)` to be initialized.
```

Comment:
Technically the only information relevant to the user is that she has to pass a struct of type `iox_service_discovery_storage_t` which will be of the correct size (by construction/API). 

Hence one could leave out the information with `sizeof`. It may give people ideas to use it with raw buffers of that size (which will work if correctly aligned but is not encouraged).

Not a must change but should probably be omitted.